packages/ai-jsx/src/lib/openai.tsx

```diff
@@ -126,6 +128,49 @@ function logitBiasOfTokens(tokens: Record<string, number>) {
   );
 }
 
+type OpenAIMethod = 'createCompletion' | 'createChatCompletion';
+type AxiosResponse<M> = M extends OpenAIMethod ? Awaited<ReturnType<InstanceType<typeof OpenAIApi>[M]>> : never;
+
+export class OpenAIError<M extends OpenAIMethod> extends HttpError {
+  readonly errorResponse: Record<string, any> | null;
+
+  constructor(response: AxiosResponse<M>, method: M, responseText: string) {
+    let errorResponse = null as Record<string, any> | null;
+    let responseSuffix = '';
+    try {
+      errorResponse = JSON.parse(responseText);
+    } catch {
+      // The response wasn't JSON, ignore it.
+    }
+
+    const parsedMessage = errorResponse?.error?.message;
+    if (typeof parsedMessage === 'string' && parsedMessage.trim().length > 0) {
+      responseSuffix = `: ${parsedMessage}`;
+    }
+
+    super(
+      `OpenAI ${method} request failed with status code ${response.status}${responseSuffix}`,
```

Comment:
Whenever this bubbles up to being read by the human, I'd like to link directly to the OpenAI dev docs.

Reply:
Good idea!